FirebaseAuthUI/AccountManagement/FUIAccountSettingsOperation.m

```diff
@@ -0,0 +1,277 @@
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#import "FUIAccountSettingsOperation_Internal.h"
+
+#import "FUIAccountSettingsOperationDeleteAccount.h"
+#import "FUIAccountSettingsOperationForgotPassword.h"
+#import "FUIAccountSettingsOperationSignOut.h"
+#import "FUIAccountSettingsOperationUnlinkAccount.h"
+#import "FUIAccountSettingsOperationUpdateEmail.h"
+#import "FUIAccountSettingsOperationUpdateName.h"
+#import "FUIAccountSettingsOperationUpdatePassword.h"
+#import "FUIAuthBaseViewController.h"
+#import "FUIAuthErrorUtils.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation FUIAccountSettingsOperation
+
++ (void)executeOperationWithDelegate:(id<FUIAccountSettingsOperationUIDelegate>)delegate
+                                 showDialog:(BOOL)showDialog {
+  [[[self alloc] initWithDelegate:delegate] execute:showDialog];
+}
+
++ (void)executeOperationWithDelegate:(id<FUIAccountSettingsOperationUIDelegate>)delegate {
+  [[[self alloc] initWithDelegate:delegate] execute:NO];
+}
+
+- (instancetype)initWithDelegate:(id<FUIAccountSettingsOperationUIDelegate>)delegate {
+  if (self = [super init]) {
+    _delegate = delegate;
+  }
+
+  return self;
+}
+
+- (void)execute:(BOOL)showDialog {
+  // This method should always be overriden.
+  [self doesNotRecognizeSelector:_cmd];
+}
+
+- (FUIAccountSettingsOperationType)operationType {
+    static NSDictionary* classMap = nil;
+
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+      classMap = @{
+        NSStringFromClass([FUIAccountSettingsOperationUpdateName class]) :
+            @(FUIAccountSettingsOperationTypeUpdateName),
+        NSStringFromClass([FUIAccountSettingsOperationUpdatePassword class]) :
+            @(FUIAccountSettingsOperationTypeUpdatePassword),
+        NSStringFromClass([FUIAccountSettingsOperationForgotPassword class]) :
+            @(FUIAccountSettingsOperationTypeForgotPassword),
+        NSStringFromClass([FUIAccountSettingsOperationUpdateEmail class]) :
+            @(FUIAccountSettingsOperationTypeUpdateEmail),
+        NSStringFromClass([FUIAccountSettingsOperationUnlinkAccount class]) :
+            @(FUIAccountSettingsOperationTypeUnlinkAccount),
+        NSStringFromClass([FUIAccountSettingsOperationSignOut class]) :
+            @(FUIAccountSettingsOperationTypeSignOut),
+        NSStringFromClass([FUIAccountSettingsOperationDeleteAccount class]) :
+            @(FUIAccountSettingsOperationTypeDeleteAccount)
+      };
+    });
+  return [classMap[NSStringFromClass([self class])] integerValue];
+}
+
+#pragma mark - protected methods
+
+- (void)finishOperationWithError:(nullable NSError *)error {
+  if (error) {
+    switch (error.code) {
+      case FIRAuthErrorCodeEmailAlreadyInUse:
+        [self showAlertWithMessage:FUILocalizedString(kStr_EmailAlreadyInUseError)];
```

Comment:
So the `authUI:didFinishOperation:error:` delegate method is not called in error case?